.github/workflows/docs.yml

```diff
@@ -0,0 +1,36 @@
+name: Docs CI
+
+on:
+  push:
+    branches:
+      - master
+      - sphinx_automation
+  workflow_dispatch:
+
+jobs:
+  build_and_run_tests:
+
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - uses: actions/checkout@v3
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: 3.9
+
+    - name: Build and install at
+      run: python -m pip install ".[plot, doc]"
+      working-directory: pyat
+
+    - name: Compile docs
+      run: make html
+      working-directory: docs
+
+    - name: Publish Docs to gh-pages
+      uses: peaceiris/actions-gh-pages@v3
```

Comment:
As described in another example [here](https://github.com/dls-controls/scanspec/blob/2136794f6898f19ec6cccb819e31ca4149191b03/.github/workflows/docs.yml), it is [recommended ](https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions#using-third-party-actions)that third-party actions are referenced by commit hash instead of tag. It is only a recommendation, but it seems reasonable.

Reply:
Hi @MJGaughran: I see the point, but I don't think we should be too much concerned about that: if we were thinking of future compatibility problems, we should rather freeze versions on numpy, matplotlib, scipy… These would cause real problems like AT crashes or wrong results. Here, such an hypothetical compatibility breaking would have negligible consequences: stopping the update of documentation for a few hours!

So no problem to set a hash instead, but I would not put the "safety" comment before, which gives the impression that we put a strong emphasis on details rather then important points

Reply:
`peaceiris/actions-gh-pages` now referenced with hash `068dc23d9710f1ba62e86896f84735d869951305` (last version)

Reply:
Late response, but hash reference is good.

This is not about future compatibility, but instead potential to run compromised action as described in the recommendation.